src/signature.c

```diff
@@ -390,10 +377,11 @@ int verify_callback(int ok, X509_STORE_CTX *stor)
 	return ok;
 }
 
-/* Downloads the corresponding signature filename from the
- * swupd server.
+/* Verifies signature for the file DATA_FILENAME, or downloads the signature
+ * first from DATA_URL and verifies the signature afterward.
```

Comment:
I think the comment on download would be better stated saying: "Verifies signature for the local file DATA_FILENAME first, and on failure downloads the signature based on DATA_URL and tries to verify again."